pallets/teerex/src/lib.rs

```diff
@@ -186,6 +205,15 @@ pub mod pallet {
 				);
 			}
 
+			// TODO: imagine this fn is not called for the first time (e.g. when worker restarts),
+			//       should we check the current sidechain_blocknumber >= registered
+			// sidechain_blocknumber?
+			#[cfg(not(feature = "skip-scheduled-enclave-check"))]
+			ensure!(
+				ScheduledEnclave::<T>::iter_values().any(|m| m == enclave.mr_enclave),
+				Error::<T>::EnclaveNotInSchedule
+			);
```

Comment:
how about skip if the ScheduledEnclave is empty?

Reply:
As discussed yesterday, we do need to set the allowed MRENCLAVE, otherwise the registration will fail.